src/sentry/web/frontend/twofactor.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import absolute_import
+
+import time
+
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+
+from sentry.web.frontend.base import BaseView
+from sentry.web.forms.accounts import TwoFactorForm
+from sentry.web.helpers import render_to_response
+from sentry.utils import auth, json
+from sentry.models import Authenticator
+
+
+class TwoFactorAuthView(BaseView):
+    auth_required = False
+
+    def perform_signin(self, request, user, interface=None):
+        auth.login(request, user, passed_2fa=True)
+        rv = HttpResponseRedirect(auth.get_login_redirect(request))
+        if interface is not None:
+            interface.authenticator.mark_used()
+            if not interface.is_backup_interface:
+                rv.set_cookie('s2fai', str(interface.type),
```

Comment:
Should we make this cookie name and duration configurable? Similar to Django's `SESSION_COOKIE_NAME`, `CSRF_COOKIE_NAME`, etc.